### PR TITLE
Some Refactoring

### PR DESCRIPTION
**Pull request by TRoZZNL**

### CODE REVIEW

File: src/Sushi.Mediakiwi.API/Controllers/BaseMediakiwiApiController.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        {
+            get
+            {
+                if (HttpContext.Items.ContainsKey(Common.API_USER_CONTEXT))
+                {
+                    return HttpContext.Items[Common.API_USER_CONTEXT] as Data.IApplicationUser;
+                }
+                else
+                {
+                    return default;
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/BaseMediakiwiApiController.cs:20]: don't use default, it hides intent. I assume this returns NULL?

File: src/Sushi.Mediakiwi.API/Controllers/Navigation.cs [hunk 2]
```diff
@@ -46,14 +44,11 @@ public async Task<ActionResult<GetTopNavigationResponse>> GetTopnavigation([From
 
             var menus = await Data.Menu.SelectAllAsync().ConfigureAwait(false);
 
-            // Extract roleID from HttpContext User
-            var roleId = Utils.ConvertToInt(User.FindFirst("roleId").Value, 0);
-
             // Get Appropriate role from DB
-            var role = await Data.ApplicationRole.SelectOneAsync(roleId).ConfigureAwait(false);
+            var role = await MediakiwiUser.RoleAsync();
```
Comment (mark-devries) [inline src/Sushi.Mediakiwi.API/Controllers/Navigation.cs:48]: rename RoleAsync() to SelectRoleAsync() (I assume it retrieves the role from database?)

**Review by mark-devries — CHANGES_REQUESTED** [review]

File: src/Sushi.Mediakiwi.API/Controllers/BaseMediakiwiApiController.cs (hunk 1 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi.API/Controllers/BaseMediakiwiApiController.cs]: Agreed, adjusted

File: src/Sushi.Mediakiwi.API/Controllers/Navigation.cs (hunk 2 above)
Comment (TRoZZNL) [reply src/Sushi.Mediakiwi.API/Controllers/Navigation.cs]: Agreed, adjusted and expanded comments on IApplicationUser

**Pull request closed without merging**